mipengine/node_tasks_DTOs.py

```diff
@@ -46,6 +58,7 @@ class TableSchema(BaseModel):
 class TableInfo(BaseModel):
     name: str
     schema_: TableSchema
+    type_: TableType
 
     _validate_identifier = validator("name", allow_reuse=True)(validate_identifier)
 
```

Comment:
The TableInfo (and we should do that for all of the dtos we have) should be immutable. 
```
class TableInfo(BaseModel):
    name: str
    schema_: TableSchema
    type_: TableType
    
    _validate_identifier = validator("name", allow_reuse=True)(validate_identifier)
    
    class Config:
        allow_mutation=False
```

Reply:
Fixed

Reply:
Nice! just one minor thing, it would be more accurate to name it ImmutableBaseModel, instead of DTOBaseModel
You could do that with a one liner
```
$ find ./ -type f -exec sed -i 's/DTOBaseModel/ImmutableBaseModel/g' {} \;
```
if you want to do it, I m just saying..